enhancements/machine-config/mco-network-configuration.md

```diff
@@ -0,0 +1,253 @@
+    ▼                                                                        │
+┌−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−┐
+╎                   Node                             ╎
+╎                                                    ╎
+╎ ┌──────────────────────────────────────┐           ╎  NodeNetworkState   ┌────────────────────────────────────────────────────────┐           MachineConfig   ┌─────────────────────────┐
+╎ │         MachineConfigDaemon          │           ╎ ──────────────────▶ │                     OpenShift-API                      │          ───────────────▶ │ MachineConfigController │
+╎ └──────────────────────────────────────┘           ╎                     └────────────────────────────────────────────────────────┘                           └─────────────────────────┘
+╎   ▲                                                ╎                       │                                  ▲                 ▲                               │
+╎   │                                                ╎                       │                                  │                 │   rendered MachineConfig      │
+╎   │ nmstatectl set/commit/rollback/show            ╎                       │ NodeNetworkConfigurationPolicy   │ MachineConfig   └───────────────────────────────┘
+╎   ▼                                                ╎                       ▼                                  │
+╎ ┌──────────────────────────────────────┐           ╎                     ┌─────────────────────────────────┐  │
+╎ │               NMState                │           ╎                     │       KNMState-Controller       │ ─┘
+╎ └──────────────────────────────────────┘           ╎                     └─────────────────────────────────┘
+╎   ▲                                                ╎
+╎   │                                                ╎
+╎   │ D-Bus                                          ╎
+╎   ▼                                                ╎
+╎ ┌──────────────────────────────────────┐           ╎
+╎ │            NetworkManager            │           ╎
+╎ └──────────────────────────────────────┘           ╎
+╎                                                    ╎
+└−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−−┘
+```
+
+### Advantages
+* Rollout keeps the same semantics as MCO. No longer a network state can end up with all the nodes losing connectivity at once.
+* Kubernetes-NMState Rollback functionality is kept (When it is triggered, it ends up in [errUnreconcilable](https://github.com/openshift/machine-config-operator/blob/15edac1b5613b135973b071cf96535c3a376419f/pkg/daemon/update.go#L193)).
+* MCO keeps ownership over node configuration.
+* API consistency and granularity for node changes is kept.
+* Customers can keep using declarative Network configuration.
+
+### Disadvantages
+* Deviates from upstream kubevirt knmstate usage. [midstream](https://github.com/openshift/kubernetes-nmstate) needs to develop:
```

Comment:
I'd add the disadvantage that the MCO team will have to maintain this, but does not have the expertise in NMState, etc... We need to make it clear who will own this going forward.

Reply:
Understood.